app/scripts/services/quota.js

```diff
@@ -0,0 +1,185 @@
+          var details;
+          if (resource.kind === 'Pod') {
+            details = "You will not be able to create the " + humanizeKind(resource.kind) + " '" + resource.metadata.name + "'.";
+          }
+          else {
+            details = "You can can still create " + humanizeKind(resource.kind) + " '" + resource.metadata.name + "' but no pods will be created until resources are freed.";
+          }
+          warnings.push({
+            type: resource.kind === 'Pod' ? 'error' : 'warning',
+            message: 'You are at your quota for ' + humanizeQuotaResource(type).toLowerCase() + ' on pods.',
+            details: details,
+            links: [{
+              href: "/project/" + quota.metadata.namespace + "/quota",
+              label: "View Quota",
+              target: "_blank"
+            }]
+          });
+        }
+      });
+
+      return warnings;
+    };
+
+    var getQuotaWarnings = function(resources, quotas, clusterQuotas) {
+      var warnings = [];
+
+      if (!resources || !quotas) {
+        return warnings;
+      }
+
+      _.each(resources, function(resource) {
+        var filteredQuotas = filterQuotasForResource(resource, quotas);
+        var filteredClusterQuotas = filterQuotasForResource(resource, clusterQuotas);
+
+        var rgv = APIService.objectToResourceGroupVersion(resource);
+        var humanizedResource = APIService.kindToResource(resource.kind, true);
+        var humanizedKind = humanizeKind(resource.kind);
+        var quotaKey = "";
+        if (rgv.group) {
+          quotaKey = rgv.group + "/";
+        }
+        quotaKey += rgv.resource;
+        var warningsForQuota = function(quota) {
+          var q = quota.status.total || quota.status;
+          if(q.hard[quotaKey] !== undefined && q.hard[quotaKey] === q.used[quotaKey]) {
+            warnings.push({
+              type: 'error',
+              message: "You are at your quota of " + q.hard[quotaKey] + " " + (q.hard[quotaKey] === "1" ? humanizedKind : humanizedResource) +
+                        " in this project.",
+              details: "You will not be able to create the " + humanizedKind + " '" + resource.metadata.name + "'.",
+              links: [{
+                href: "/project/" + quota.metadata.namespace + "/quota",
+                label: "View Quota",
+                target: "_blank"
+              }]
+            });
+          }
+
+          warnings = warnings.concat(getResourceLimitWarnings(resource, quota));
+        };
+        _.each(filteredQuotas, function(quota) {
+          warningsForQuota(quota);
+        });
+        _.each(filteredClusterQuotas, function(quota) {
+          warningsForQuota(quota);
+        });
+      });
+
+      return warnings;
+    };
+
+    var getLatestQuotaWarnings = function(resources, context) {
+      var result = $q.defer();
+      var quotaWarnings = [];
+      var quotas, clusterQuotas;
+      var remaining = 2;
+      function _checkDone() {
```

Comment:
I actually don't want to switch to $q.all because of the issue with it rejecting on the first rejection, I prefer we actually wait for all rejections to happen before we redirect.  I was actually planning on switching the $q.all thats in from yaml/json